sambacc/commands/run.py

```diff
@@ -47,6 +77,17 @@ def _run_container_args(parser):
             " The special 'init-all' name will perform all known setup steps."
         ),
     )
+    _wait_for_choices = _wait_for_conditions.keys()
+    parser.add_argument(
+        "--wait-for",
+        action="append",
+        choices=_wait_for_choices,
+        help=(
+            "Specify a condition to wait for prior to starting the server"
```

Comment:
FYI: `ctdb event script enable legacy 50.samba`

CTDB can manage Samba services(_nmb_ for `nmbd`, _smb_ for `smbd`, _winbind_ for `winbindd` etc) with the help of event scripts as mentioned in the [docs](https://ctdb.samba.org/manpages/ctdb-script.options.5.html) under "SERVICE CONFIGURATION". This is in fact currently in use by sit-environment to setup [clustered samba](https://github.com/samba-in-kubernetes/sit-environment/blob/main/playbooks/ansible/roles/ctdb.setup/tasks/main.yml#L96).

Reply:
I know, but we use containers and this script is not appropriate for that environment, it uses the `service` command. This is one of the things I complain about when I say that ctdb was not designed for containers and interoperating with a higher level of orchestration. IMO, most of the event scripts are not appropriate for containers.

In ceph orch I was looking at two choices... either make the primary container variable (switch the primary between ctdb and smbd) depending on deployment type or always having smbd be the primary and then making it delay the start. I went with the latter for consistentcy when debugging and thinking about the system but I needed a slightly cleaner way for waiting to ctdb to be ready than just delaying from a fixed length of time.

Reply:
I completely get it, thanks.